src/envoy/mixer/mixer_control.cc

```diff
@@ -59,28 +63,49 @@ void CreateEnvironment(Upstream::ClusterManager& cm,
 HttpMixerControl::HttpMixerControl(const HttpMixerConfig& mixer_config,
                                    Upstream::ClusterManager& cm,
                                    Event::Dispatcher& dispatcher,
-                                   Runtime::RandomGenerator& random)
-    : cm_(cm) {
+                                   Runtime::RandomGenerator& random,
+                                   const std::string& stats_prefix,
+                                   Stats::Scope& scope)
+    : cm_(cm), stats_obj_(dispatcher, stats_prefix, scope) {
   ::istio::mixer_control::http::Controller::Options options(
       mixer_config.http_config, mixer_config.legacy_quotas);
 
   CreateEnvironment(cm, dispatcher, random, &options.env);
 
   controller_ = ::istio::mixer_control::http::Controller::Create(options);
 
+  stats_obj_.InitGetStatisticsFunc(
+      std::bind(&::istio::mixer_control::http::Controller::GetStatistics,
+                controller(), std::placeholders::_1));
+
+  // Initialize old_stats for envoy stats update.
+  stats_obj_.GetStatistics(stats_obj_.mutate_old_stats());
+  stats_obj_.SetUpStatsTimer();
+
   has_v2_config_ = mixer_config.has_v2_config;
 }
 
 TcpMixerControl::TcpMixerControl(const TcpMixerConfig& mixer_config,
                                  Upstream::ClusterManager& cm,
                                  Event::Dispatcher& dispatcher,
-                                 Runtime::RandomGenerator& random) {
+                                 Runtime::RandomGenerator& random,
+                                 const std::string& stats_prefix,
+                                 Stats::Scope& scope)
+    : stats_obj_(dispatcher, stats_prefix, scope) {
   ::istio::mixer_control::tcp::Controller::Options options(
       mixer_config.tcp_config);
 
   CreateEnvironment(cm, dispatcher, random, &options.env);
 
   controller_ = ::istio::mixer_control::tcp::Controller::Create(options);
+
+  stats_obj_.InitGetStatisticsFunc(
+      std::bind(&::istio::mixer_control::tcp::Controller::GetStatistics,
+                controller(), std::placeholders::_1));
```

Comment:
please use lambda function, such as

do it in the consturctor.

stats_obj_(dispatcher, "tcp_mixer_filter.", scope,  
                 [this](STat *stat) -> bool {
                  if (!controller_) return false;
                  controler_->GetStats(stat);
                   return true;
              });